spring/boot3-autoconfigure/src/main/java/com/linecorp/armeria/internal/spring/ArmeriaConfigurationUtil.java

```diff
@@ -124,7 +124,8 @@ public static void configureServerWithArmeriaSettings(
         final boolean needsManagementPort =
                 findBean(beanFactory, "actuatorServerConfigurator", ArmeriaServerConfigurator.class) != null;
 
-        if (needsManagementPort && managementServerPort != null) {
+        if (needsManagementPort && managementServerPort != null &&
+            (internalServicePort == null || internalServicePort.getPort() != managementServerPort.getPort())) {
```

Comment:
What do you think of using a set instead of a list for `internalPortsBuilder` and implementing the `equals` method in the `Port`?
This is only checking the port number and it might have a different `InetAddress` so I think it's safe to use a set for comparing the `Port` completely.

We can also change the signature of `configurePorts` method:
```java
public final class ArmeriaConfigurationNetUtil {
    public static void configurePorts(ServerBuilder server, Set<Port> ports) {...}
}

configurePorts(server, ImmutableSet.copyOf(settings.getPorts()));
```

Reply:
I think it's good idea! In fact, I initially wrote the code using `ImmutableSet`. I defined the criteria for determining if two `Port` objects are equal based solely on the `int port` field and created `equals` and `hashCode` methods accordingly. However, since `Port` has many other fields, I realized that using such a simple criterion to consider two `Port` objects as equal might cause issues later on. So, I'm planning to use `ImmutableMap` instead of `ImmutableSet`, and modify the code by setting the key as int port.

Reply:
Oops, I realized that we cannot use a set because we need to compare only `Port.getPort()`, `Port.getAddress()`, and `Port.getIp()` for equality. (We might merge `Port.getProtocols()` if two `Port`s have different values.)
So how about using the list as is and just deduping the `Port`?
```java
configurePorts(server, dedupPorts(internalPorts));

private static List<Port> dedupPorts(List<Port> ports) {
    // return the duduped ports.
}
```

Reply:
Thank you for confirming. 😄 I have been thinking about how to handle the equality checking logic, and the above method is a good approach. I will apply it and push the changes.